doc/source/development/dispatching.rst

```diff
@@ -0,0 +1,93 @@
+Dispatching
+===========
+
+The dispatching mechanism allows users to use alternative implementations of the algorithms
+available in scikit-image. The alternative implementations are provided by separate
+Python packages maintained outside of scikit-image. Example use-cases are support for array
+libraries other than Numpy or optimised implementations for particular hardware.
+
+This section of the documentation describes how to create an alternative implementation (a "backend")
+and how the dispatching mechanism in scikit-image works.
+
+.. important::
+    Dispatching support and infrastructure is experimental and is not ready for production.
+    It's made available as an early draft so that developers can gain experience
+    with the system.
+
+    Expect the dispatching API to change without notice.
+
+
+Implementing a new backend
+--------------------------
+
+An alternative implementation ("backend") is a good place to provide optimized implementations
+for particular hardware, support array libraries other than Numpy or explore novel ideas that
+are not (yet) a good fit for the core scikit-image library.
+
+To create a backend you have to create a new Python package that registers two particular
+`entrypoints <https://packaging.python.org/en/latest/specifications/pyproject-toml/#entry-points>`_. The first entrypoint called `skimage_backends` should resolve to a namespace
+that contains two functions: `can_has(name, *args, **kwargs)` and `get_implementation(name)`.
+The second entrypoint called `skimage_backend_infos` should resolve to a function with no
+arguments that returns an instance of the `skimage.util._backends.BackendInformation` class
+or something that behaves like it.
```

Comment:
I like the idea of linking it, but in my local build of the docs the API reference doesn't seem to get built? Maybe @lagru has an idea on how to do it?

I'd rather not write more here, it should be in the doc string of the class and we should link to it.

I changed up the two sentences referring to the two entry points to link to the respective sections.

Reply:
> > Also, it would be useful to properly describe the behaviour(like the parameters, return, etc.) of the can_has and get_implementation. We can add that in the following PRs as well.

> can_has and get_implementation are described in more detail in the section describing the skimage_backends entry point

Describing these functions' behaviour like we describe all other algorithms (i.e. following the `sections` guide in the numpydoc style guide - https://numpydoc.readthedocs.io/en/latest/format.html#sections) would make it easier for others to understand it and also it would be easier to change the documentation in future that way, and it would also be consistent with how most people are used to seeing a function's behaviour. But, this can be left for future PRs.